packages/fxa-auth-server/test/local/payments/subscription-reminders.js

```diff
@@ -0,0 +1,389 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+const { assert } = require('chai');
+const sinon = require('sinon');
+const { Container } = require('typedi');
+
+// eslint-disable-next-line no-unused-vars
+const error = require('../../../lib/error');
+const { mockLog } = require('../../mocks');
+const { CurrencyHelper } = require('../../../lib/payments/currencies');
+const { StripeHelper } = require('../../../lib/payments/stripe');
+const { SentEmail } = require('fxa-shared/db/models/auth/sent-email');
+const {
+  EMAIL_TYPE,
+  SubscriptionReminders,
+  MS_IN_A_DAY,
+} = require('../../../lib/payments/subscription-reminders');
+const {
+  DEFAULT_PLAN_LENGTH,
+  DEFAULT_REMINDER_LENGTH,
+} = require('../../../scripts/subscription-reminders');
+const longPlan1 = require('./fixtures/stripe/plan1.json');
+const longPlan2 = require('./fixtures/stripe/plan2.json');
+const shortPlan1 = require('./fixtures/stripe/plan3.json');
+const longSubscription1 = require('./fixtures/stripe/subscription1.json'); // sub to plan 1
+const longSubscription2 = require('./fixtures/stripe/subscription2.json'); // sub to plan 2
+const planLength = 30; // days
+const reminderLength = 14; // days
+
+const sandbox = sinon.createSandbox();
+
+/**
+ * To prevent the modification of the test objects loaded, which can impact other tests referencing the object,
+ * a deep copy of the object can be created which uses the test object as a template
+ *
+ * @param {Object} object
+ */
+function deepCopy(object) {
+  return JSON.parse(JSON.stringify(object));
+}
+
+describe('SubscriptionReminders', () => {
+  let mockStripeHelper;
+  let reminder;
+  let mockConfig;
+
+  beforeEach(() => {
+    mockConfig = {
+      currenciesToCountries: { ZAR: ['AS', 'CA'] },
+    };
+    mockStripeHelper = {
+      findActiveSubscriptionsByPlanId: () => {},
+    };
+    const currencyHelper = new CurrencyHelper(mockConfig);
+    Container.set(CurrencyHelper, currencyHelper);
+    Container.set(StripeHelper, mockStripeHelper);
+    reminder = new SubscriptionReminders(
+      mockLog,
+      mockConfig,
+      planLength,
+      reminderLength,
+      {},
+      {}
+    );
+  });
+
+  afterEach(() => {
+    sandbox.reset();
+  });
+
+  describe('constructor', () => {
+    it('sets log, planLength, reminderLength and stripe helper', () => {
+      assert.strictEqual(reminder.log, mockLog);
+      assert.equal(reminder.planLength, planLength);
+      assert.equal(reminder.reminderLength, reminderLength);
+      assert.strictEqual(reminder.stripeHelper, mockStripeHelper);
+    });
+    it('uses defaults for planLength and reminderLength if no values are passed for them', () => {
+      const reminder = new SubscriptionReminders(mockLog, mockConfig);
+      assert.equal(reminder.planLength, DEFAULT_PLAN_LENGTH);
+      assert.equal(reminder.reminderLength, DEFAULT_REMINDER_LENGTH);
```

Comment:
huh ok, my comment about the default arguments not auto set into the object properties was totally wrong.  I've never used that syntax sugar before.

Reply:
Yeah it seemed to work fine. I made it explicit anyway.